code/parachain/frame/assets-transactor-router/src/tests/routes.rs

```diff
@@ -0,0 +1,1978 @@
+			<Pallet<Test> as Unbalanced<AccountId>>::set_total_issuance(asset_id_foreign, 2000);
+
+			assert_eq!(<<Test as Config>::NativeTransactor>::total_issuance(), 3000);
+			assert_eq!(<<Test as Config>::LocalTransactor>::total_issuance(asset_id_local), 1000);
+			assert_eq!(
+				<<Test as Config>::ForeignTransactor>::total_issuance(asset_id_foreign,),
+				2000
+			);
+		});
+	}
+	#[test]
+	fn transfer() {
+		let protocol_id_local = *b"testloca";
+		let nonce_local = 0;
+		let protocol_id_foreign = *b"testfore";
+		let nonce_foreign = 0;
+		let asset_info_local = AssetInfo {
+			name: Some(
+				BiBoundedAssetName::from_vec(b"local asset".to_vec())
+					.expect("string is within bound"),
+			),
+			symbol: None,
+			decimals: Some(12),
+			ratio: None,
+			existential_deposit: 100,
+		};
+
+		new_test_ext().execute_with(|| {
+			let asset_id_local = Pallet::<Test>::create_local_asset(
+				protocol_id_local,
+				nonce_local,
+				asset_info_local,
+			)
+			.unwrap();
+
+			let foreign_asset_id = XcmAssetLocation(xcm::v2::MultiLocation::parent());
+			let foreign_asset_info = AssetInfo {
+				name: Some(
+					BiBoundedAssetName::from_vec(b"Kusama".to_vec())
+						.expect("string is within bound"),
+				),
+				symbol: Some(
+					BiBoundedAssetSymbol::from_vec(b"KSM".to_vec())
+						.expect("string is withing bound"),
+				),
+				decimals: Some(12),
+				ratio: None,
+				existential_deposit: 1000,
+			};
+
+			let asset_id_foreign = Pallet::<Test>::create_foreign_asset(
+				protocol_id_foreign,
+				nonce_foreign,
+				foreign_asset_info,
+				foreign_asset_id,
+			)
+			.unwrap();
+
+			Pallet::<Test>::mint_into(RuntimeOrigin::root(), asset_id_local, ACCOUNT_LOCAL, 1000)
+				.unwrap();
+			Pallet::<Test>::mint_into(
+				RuntimeOrigin::root(),
+				asset_id_foreign,
+				ACCOUNT_FOREIGN,
+				2000,
+			)
+			.unwrap();
+			Pallet::<Test>::mint_into(RuntimeOrigin::root(), NATIVE_ASSET_ID, ACCOUNT_NATIVE, 3000)
+				.unwrap();
+
+			assert_ok!(<Pallet::<Test> as Transfer<AccountId>>::transfer(
+				NATIVE_ASSET_ID,
+				&ACCOUNT_NATIVE,
+				&ACCOUNT_TO,
+				3000,
+				false
+			));
+			assert_ok!(<Pallet::<Test> as Transfer<AccountId>>::transfer(
+				asset_id_local,
+				&ACCOUNT_LOCAL,
+				&ACCOUNT_TO,
+				1000,
+				false
+			));
+			assert_ok!(<Pallet::<Test> as Transfer<AccountId>>::transfer(
+				asset_id_foreign,
+				&ACCOUNT_FOREIGN,
+				&ACCOUNT_TO,
+				2000,
+				false
+			));
+			assert_eq!(
+				<Pallet::<Test> as Inspect<AccountId>>::balance(NATIVE_ASSET_ID, &ACCOUNT_TO,),
+				3000
+			);
+			assert_eq!(
+				<Pallet::<Test> as Inspect<AccountId>>::balance(asset_id_local, &ACCOUNT_TO,),
+				1000
+			);
+			assert_eq!(
+				<Pallet::<Test> as Inspect<AccountId>>::balance(asset_id_foreign, &ACCOUNT_TO,),
+				2000
+			);
+
+			assert_eq!(<<Test as Config>::NativeTransactor>::total_balance(&ACCOUNT_TO), 3000);
+			assert_eq!(
+				<<Test as Config>::LocalTransactor>::total_balance(asset_id_local, &ACCOUNT_TO),
```

Comment:
is the full `<<Test as Config>::LocalTransactor>` required? or does `Test::LocalTransactor` work?

Reply:
error message for second option returns "use fully-qualified syntax `<mocks::Test as Trait>::LocalTransactor`"

Reply:
Perhaps some type aliases `type LocalTransactorOf<T> = <T as Config>::LocalTransactor>;` would make it more concise